client/src/app/+videos/+video-edit/shared/thumbnail-manager/thumbnail-manager.component.html

```diff
@@ -0,0 +1,29 @@
+<div class="root">
+    <div *ngIf="!selectingFromVideo" class="preview-container">
+  
+      <img *ngIf="imageSrc" [ngStyle]="{ width: previewWidth, height: previewHeight }" [src]="imageSrc" class="preview"
+        alt="Preview" i18n-alt />
+      <div *ngIf="!imageSrc" [ngStyle]="{ width: previewWidth, height: previewHeight }" class="preview no-image"></div>
+  
+    </div>
+  
+    <div id="embedContainer" class="" style="position: relative;"></div>
```

Comment:
You can use and improve our embed component instead of injecting your embed code? https://github.com/Chocobozzz/PeerTube/blob/develop/client/src/app/shared/shared-main/video/embed.component.ts

Reply:
I made a comment above about this. I made an honest effort to make the my-embed component work, however, I couldn't get the Embed API to work with it. I did search the code for other places it was used but couldn't find anything. If there is something I have missed or if you have some thoughts on how to approach, I can try again.